src/Raven.Server/Documents/Handlers/IndexHandler.cs

```diff
@@ -652,6 +660,19 @@ public Task Terms()
             }
         }
 
+        private string GetIndexNameFromCollectionAndField(string field, DocumentsOperationContext context)
+        {
+            var collection = GetStringQueryString("collection", false);
+            if (string.IsNullOrEmpty(collection))
+                ThrowRequiredMember("name");
+            var query = new IndexQueryServerSide(new QueryMetadata(collection, field));
+            var dynamicQueryToIndex = new DynamicQueryToIndexMatcher(Database.IndexStore);
+            var match = dynamicQueryToIndex.Match(DynamicQueryMapping.Create(query), context);
+            if (match.MatchType == DynamicQueryMatchType.Failure)
```

Comment:
I think that it needs to be a complete match?

Reply:
Why? In partial match I see we clone the index... so it means the index does not have the requested field?

Reply:
In particular, I don't think you _can_ get partial here.
You are only ever sending a single field, after all.